code/datastructure-set/src/test/java/ru/unn/agile/datastructure/set/model/MySetTest.java

```diff
@@ -0,0 +1,244 @@
+package ru.unn.agile.datastructure.set.model;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+public class MySetTest {
+    @Test
+    public void canInitDefaultConstructor() {
+        MySet mySet = new MySet();
+        Assert.assertNotNull(mySet);
+    }
+
+    @Test
+    public void canInitWithSpecificType() {
+        MySet<String> mySet = new MySet<>();
+        Assert.assertNotNull(mySet);
+    }
+
+    @Test
+    public void canGetZeroSize() {
+        MySet mySet = new MySet();
+        Assert.assertEquals(mySet.size(), 0);
+    }
+
+    @Test
+    public void canAddElement() {
+        MySet<String> mySet = new MySet<>();
+        Assert.assertTrue(mySet.add("test"));
+    }
+
+    @Test
+    public void canGetRealSize() {
+        MySet<String> mySet = new MySet<>();
+        mySet.add("first row");
+        mySet.add("second row");
+        Assert.assertEquals(mySet.size(), 2);
+    }
+
+    @Test
+    public void canCheckEmpty() {
+        MySet mySet = new MySet();
+        Assert.assertTrue(mySet.isEmpty());
+    }
+
+    @Test
+    public void canCheckNotEmpty() {
+        MySet<String> mySet = new MySet<>();
+        mySet.add("test value");
+        Assert.assertFalse(mySet.isEmpty());
+    }
+
+    @Test
+    public void canBeCleared() {
+        MySet<String> mySet = new MySet<>();
+        mySet.add("first row");
+        mySet.add("second row");
+        mySet.clear();
+        Assert.assertTrue(mySet.isEmpty());
+    }
+
+    @Test
+    public void canConvertToArray() {
+        MySet<String> mySet = new MySet<>();
+        mySet.add("first");
+        mySet.add("second");
+        Assert.assertArrayEquals(new String[]{"first", "second"}, mySet.toArray());
+    }
+
+    @Test
+    public void canNotAddDuplicateElement() {
+        MySet<String> mySet = new MySet<>();
+        mySet.add("first");
+        mySet.add("first");
+        Assert.assertEquals(mySet.size(), 1);
```

Comment:
Please change parameters order: the first parameter must be an expected result, and the second one - actual

Reply:
Please read the test case name :)

Reply:
Agree fixed.